utils/utils.jsx

```diff
@@ -138,32 +139,57 @@ export function getTeamRelativeUrl(team) {
+    } else if (channel) {
+        const team = getTeam(state, teamId);
+        notificationURL = getTeamRelativeUrl(team) + '/channels/' + channel.name;
+    } else if (teamId) {
+        const team = getTeam(state, teamId);
+        const redirectChannel = getRedirectChannelNameForTeam(state, teamId);
+        notificationURL = getTeamRelativeUrl(team) + `/channels/${redirectChannel}`;
+    } else {
+        const currentTeamId = getCurrentTeamId(state);
+        const redirectChannel = getRedirectChannelNameForTeam(state, currentTeamId);
+        notificationURL = getCurrentRelativeTeamUrl(state) + `/channels/${redirectChannel}`;
+    }
+    return notificationURL;
+}
+
 export function notifyMe(title, body, channel, teamId, silent) {
-    showNotification({title,
-        body,
-        requireInteraction: false,
-        silent,
-        onClick: () => {
-            const state = store.getState();
-            window.focus();
-            if (channel && (channel.type === Constants.DM_CHANNEL || channel.type === Constants.GM_CHANNEL)) {
-                browserHistory.push(getCurrentRelativeTeamUrl(state) + '/channels/' + channel.name);
-            } else if (channel) {
-                const team = getTeam(state, teamId);
-                browserHistory.push(getTeamRelativeUrl(team) + '/channels/' + channel.name);
-            } else if (teamId) {
-                const team = getTeam(state, teamId);
-                const redirectChannel = getRedirectChannelNameForTeam(state, teamId);
-                browserHistory.push(getTeamRelativeUrl(team) + `/channels/${redirectChannel}`);
-            } else {
-                const currentTeamId = getCurrentTeamId(state);
-                const redirectChannel = getRedirectChannelNameForTeam(state, currentTeamId);
-                browserHistory.push(getCurrentRelativeTeamUrl(state) + `/channels/${redirectChannel}`);
-            }
-        },
-    }).catch(() => {
-        // Ignore the failure to display the notification.
-    });
+    // handle notifications in desktop app >= 4.3.0
+    if (UserAgent.isDesktopApp() && window.desktop && semver.gte(window.desktop.version, '4.3.0')) {
+        // get the desktop app to trigger the notification
+        window.postMessage(
+            {
+                type: 'dispatch-notification',
+                message: {
+                    title,
+                    body,
+                    channel,
+                    teamId,
+                    silent,
+                },
+            },
+            window.location.origin
+        );
+    } else {
+        showNotification({
+            title,
+            body,
+            requireInteraction: false,
+            silent,
+            onClick: () => {
+                window.focus();
+                browserHistory.push(getChannelURL(channel, teamId));
+            },
+        }).catch(() => {
+            // Ignore the failure to display the notification.
```

Comment:
do we use sentry for the desktop? this would be a good scenario to get what happened.

At least, we should probably log it somewhere. At some point a user is going to not receive notifications and we are going to have a harder time finding why.

Reply:
Agreed. I can log failures on the Desktop App side. I'm not currently familiar with how that is done here in the Webapp (that catch statement is existing code, just moved a bit). I'll see if I can find out.

Reply:
Done